src/System.Windows.Forms/src/System/Windows/Forms/Control.cs

```diff
@@ -11523,16 +11530,36 @@ protected internal void UpdateBounds()
             int clientWidth = rect.right;
             int clientHeight = rect.bottom;
             User32.GetWindowRect(new HandleRef(_window, InternalHandle), ref rect);
+
+            int width = rect.right - rect.left;
+            int height = rect.bottom - rect.top;
+
             if (!GetTopLevel())
             {
                 User32.MapWindowPoints(IntPtr.Zero, User32.GetParent(new HandleRef(_window, InternalHandle)), ref rect);
+
+                // When moving the Form to a new monitor with different DPI settings, Windows changing the Dpi
+                // on the handle and sendign DPI changed message to the Form are not in sync. Due to this,
+                // methods User32.GetClientRect are returnign the incorrect value.  This method is not
+                // accurate with respect to adornments metrics and their Dpi. Hence this work around.
```

Comment:
I'm finding "_This method is not accurate_" confusing, I can't infer from the context which method it refers to
```suggestion
                // methods User32.GetClientRect are returnign the incorrect value.  This method is not
                // accurate with respect to adornments metrics and their Dpi. Hence this work around.
```